### PR TITLE
Support language auto-detection

### PR DESCRIPTION
**Pull request by Hengyu**
This pull request contains following improvements:
- Fixed error #6 
- Uses `languageAuto` function in `highlight.js` to auto detect the programming languages. (This is useful if you can't determine the programming language via the file extension, eg: ".h")

### CODE REVIEW

**Comment by raspu** [comment]
Hi! Thank you for taking the time to do this, I appreciate it. I have some comments that I will write in the diff itself.

File: Pod/Classes/Highlightr.swift [hunk 1]
```diff
@@ -78,50 +78,54 @@ open class Highlightr
...
-     */
-    open func highlight(_ code: String, as languageName: String, fastRender: Bool = true) -> NSAttributedString?
+    /// Takes a String and returns a NSAttributedString with the given language highlighted.
+    ///
+    /// - Parameters:
+    ///   - code: Code to highlight.
+    ///   - languageName: Language name or alias.
+    ///   - needsFastRender: Defaults to true - When *true* will use the custom made html parser rather than Apple's solution.
+    /// - Returns: NSAttributedString with the detected code highlighted.
+    open func highlight(_ code: String, language languageName: String? = nil, needsFastRender: Bool = true) -> NSAttributedString?
```
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: You changed the highlight method signature (from as: to language:), this was renamed recently in an attempt of following the newer Swift 3 style of less verbose methods, and I prefer to keep it this way. Would it be possible to revert this change in your PR?

File: Pod/Classes/Highlightr.swift [hunk 2]
```diff
@@ -78,50 +78,54 @@ open class Highlightr
...
-     - returns: NSAttributedString with the detected code highlighted.
-     */
-    open func highlight(_ code: String, as languageName: String, fastRender: Bool = true) -> NSAttributedString?
+    /// Takes a String and returns a NSAttributedString with the given language highlighted.
+    ///
+    /// - Parameters:
+    ///   - code: Code to highlight.
+    ///   - languageName: Language name or alias.
+    ///   - needsFastRender: Defaults to true - When *true* will use the custom made html parser rather than Apple's solution.
+    /// - Returns: NSAttributedString with the detected code highlighted.
```
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: Please use /** */ for multi-line docs.

**Comment by Hengyu** [comment]
You are welcome :)

File: Pod/Classes/Highlightr.swift [hunk 3]
```diff
@@ -14,7 +14,7 @@ open class Highlightr
 {
     /// Returns the current Theme.
     open var theme : Theme!
-    {
+        {
```
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: I am following Allman style for the indentation, would you please revert this?

File: Pod/Classes/Highlightr.swift (hunk 1 above)
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: Also "needsFastRender" makes the option more confusing, like if some code may or not need fast render. I am not really happy with fastRender to be honest, but I like it better that needsFastRender.

File: Pod/Classes/Highlightr.swift (hunk 2 above)
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: Also, are you sure if this format (Parameters: as a list, and both Parameters and Returns capitalized) is supported by [jazzy](https://github.com/realm/jazzy) or any other Swift doc generator?

File: Pod/Classes/Highlightr.swift [hunk 4]
```diff
@@ -78,50 +78,54 @@ open class Highlightr
...
         var fixedCode = code.replacingOccurrences(of: "\\",with: "\\\\");
         fixedCode = fixedCode.replacingOccurrences(of: "\'",with: "\\\'");
         fixedCode = fixedCode.replacingOccurrences(of: "\"", with:"\\\"");
         fixedCode = fixedCode.replacingOccurrences(of: "\n", with:"\\n");
         fixedCode = fixedCode.replacingOccurrences(of: "\r", with:"");
-
-        let command =  String.init(format: "%@.highlight(\"%@\",\"%@\").value;", hljs,languageName, fixedCode)
+        
+        let command: String
+        if let languageName = languageName {
```
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: Please use Allman indentation here also.

File: Pod/Classes/Highlightr.swift [hunk 5]
```diff
@@ -78,50 +78,54 @@ open class Highlightr
...
         let res = jsContext.evaluateScript(command)
         guard var string = res!.toString() else
         {
             return nil
         }
         
         let returnString : NSAttributedString
-        if(fastRender)
-        {
+        if(needsFastRender) {
```
Comment (raspu) [inline Pod/Classes/Highlightr.swift]: Please use Allman indentation here also.

**Comment by raspu** [comment]
There you go, most of them are formatting. Sorry if I am being picky, but I would like to keep the code consistent.

File: Pod/Classes/Highlightr.swift (hunk 2 above)
Comment (Hengyu) [inline Pod/Classes/Highlightr.swift]: I'm not really sure about it. I used the default doc generating method in Xcode (Shift+Command+/). I'll revert it to the original style.

**Comment by Hengyu** [comment]
Sorry about that, I ignored the original formatting preferences when I changed the code. I'll change the style back to its original.

**Comment by raspu** [comment]
Thanks! 👍

**Comment by Hengyu** [comment]
@raspu Please check the latest commit :)

**Comment by raspu** [comment]
That's perfect :). Thank you!
